### PR TITLE
[ENH] Add `SurfaceImage.from_volume`

### LINKED ISSUE #4587: [API] SurfaceImage constuct from a volume image
Briefly discussed with @jeromedockes.

Currently the SurfaceImage can do 

```python
img = SurfaceImage(mesh, data: PolyData | dict[str, Mesh | str | Path] | Nifti1Image | str | Path)
```

if Nifti1Image | str | Path is passed then `vol_to_surf` will be used under the hood to project nifti image on the mesh.

But none of the options of `vol_to_surf` can be passed as extra parameters to the constructor of SurfaceImage.

So maybe it'd be better to have:

```python
img = SurfaceImage(mesh, data: PolyData | dict[str, Mesh | str | Path])
img = SurfaceImage.from_volume(mesh, data: Nifti1Image | str | Path, vol_to_surf_kwargs: dict[str, Any])
```

### PR DESCRIPTION
**Pull request by jeromedockes**
Add a class method to construct a SurfaceImage from volume rather than using the constructor

- closes #4587

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
Nice, looking forward to it !

File: nilearn/experimental/surface/_surface_image.py [hunk 1]
```diff
@@ -191,37 +191,85 @@ class SurfaceImage:
...
         _check_data_and_mesh_compat(self.mesh, self.data)
 
         self.shape = self.data.shape
 
+    @classmethod
+    def from_volume(
+        cls, mesh, volume_img, inner_mesh=None, **vol_to_surf_kwargs
+    ):
+        """Create surface image from volume image.
+
```
Comment (bthirion) [inline nilearn/experimental/surface/_surface_image.py:226]: Can you write the docstring ?

Comment (Remi-Gau) [reply nilearn/experimental/surface/_surface_image.py:225]: added one

**Review by man-shu — APPROVED** [review]
Looks good!

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
thanks @jeromedockes for starting this
thanks @bthirion for the cross check

**Pull request merged**